march_hardware/src/EtherCAT/EthercatMaster.cpp

```diff
@@ -41,6 +41,14 @@ int EthercatMaster::getCycleTime() const
   return this->cycle_time_ms_;
 }
 
+void EthercatMaster::waitForPdo()
+{
+  std::unique_lock<std::mutex> lock(this->wait_on_pdo_condition_mutex_);
+  this->wait_on_pdo_condition_var_.wait(lock, [&] { return this->pdo_received_; });
+  this->pdo_received_ = false;
+  lock.unlock();
```

Comment:
This unlock is not necessary since [`std::unique_lock`](http://www.cplusplus.com/reference/mutex/unique_lock/) automatically releases the lock when destroyed

Reply:
Oh you are right, I misinterpreted the example.